### PR TITLE
MAINT: Cleanup remaining PyUString_ConcatAndDel use.

### PR DESCRIPTION
**Pull request by charris**
The `build_shape_string` function has also been removes as a duplicate of `convert_shape_to_string`.

This should probably be squash merged.

### CODE REVIEW

File: numpy/core/src/common/array_assign.c [hunk 1]
```diff
@@ -64,19 +64,22 @@ broadcast_strides(int ndim, npy_intp const *shape,
...
-                build_shape_string(ndim, shape));
-        PyErr_SetObject(PyExc_ValueError, errmsg);
-        Py_DECREF(errmsg);
+        PyObject *shape1 = convert_shape_to_string(strides_ndim,
+                                                   strides_shape, "");
+        if (shape1 == NULL) {
+            return -1;
+        }
 
+        PyObject *shape2 = convert_shape_to_string(ndim, shape, "");
```
Comment (eric-wieser) [inline numpy/core/src/common/array_assign.c:73]: As a follow-up to this collection of PRs, its very temping to remove the third argument from this function

File: numpy/core/src/umath/ufunc_object.c [hunk 2]
```diff
@@ -5997,18 +5998,19 @@ ufunc_get_doc(PyUFuncObject *ufunc)
...
      * of it the doc string shouldn't need the calling convention
      */
-    doc = PyObject_CallFunctionObjArgs(
-        _sig_formatter, (PyObject *)ufunc, NULL);
+    doc = PyObject_CallFunctionObjArgs(_sig_formatter,
+                                       (PyObject *)ufunc, NULL);
     if (doc == NULL) {
         return NULL;
     }
+    // FIXME: check if ufunc docstrings can/should be UTF-8
```
Comment (eric-wieser) [inline numpy/core/src/umath/ufunc_object.c:6006]: I don't understand the quetion behind this comment - the PR hasn't changed anything, has it?

**Review by eric-wieser — APPROVED** [review]
Looks good, let's remove that `FIXME` though

File: numpy/core/src/umath/ufunc_object.c (hunk 2 above)
Comment (eric-wieser) [reply numpy/core/src/umath/ufunc_object.c:6006]:
```suggestion
```

File: numpy/core/src/common/array_assign.c (hunk 1 above)
Comment (charris) [reply numpy/core/src/common/array_assign.c:73]: Good idea. It doesn't get used and is easy enough to add elsewhere if needed.

File: numpy/core/src/umath/ufunc_object.c (hunk 2 above)
Comment (charris) [reply numpy/core/src/umath/ufunc_object.c]: Done.

**Review by seberg — APPROVED** [review]
Well, tried to see if I find anything that might trip PyPy, but didn't manage. On the up-side, the code changes look very nice :).

**Pull request merged**